frame.py

```diff
@@ -0,0 +1,225 @@
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(self.G, self.LY, self.LZ), FreeCAD.Rotation(FreeCAD.Vector(0,1,0),90), FreeCAD.Vector(0,0,0))
+		# Finally add y pipes.
+		tmp = Draft.clone(ypipe)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(self.LX, self.G, 0), FreeCAD.Rotation(FreeCAD.Vector(1,0,0),-90), FreeCAD.Vector(0,0,0))
+		tmp = Draft.clone(ypipe)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(0, self.G, self.LZ), FreeCAD.Rotation(FreeCAD.Vector(1,0,0),-90), FreeCAD.Vector(0,0,0))
+		tmp = Draft.clone(ypipe)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(self.LX, self.G, self.LZ), FreeCAD.Rotation(FreeCAD.Vector(1,0,0),-90), FreeCAD.Vector(0,0,0))
+
+	def addCorners(self, group, convertToSolid):
+		corner = self.corner.create(convertToSolid)
+		group.addObject(corner)
+		# clone the corners and put them on right positions.
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(0, 0, self.LZ), FreeCAD.Rotation(FreeCAD.Vector(0,1,0),90), FreeCAD.Vector(0,0,0))
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(self.LX,0, self.LZ), FreeCAD.Rotation(FreeCAD.Vector(0,1,0),180), FreeCAD.Vector(0,0,0))
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(self.LX,0,0), FreeCAD.Rotation(FreeCAD.Vector(0,1,0),270), FreeCAD.Vector(0,0,0))
+
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(self.LX,self.LY,0), FreeCAD.Rotation(FreeCAD.Vector(0,0,1),180), FreeCAD.Vector(0,0,0))
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(0,self.LY,0), FreeCAD.Rotation(FreeCAD.Vector(0,0,1),270), FreeCAD.Vector(0,0,0))
+
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(0,self.LY, self.LZ), FreeCAD.Rotation(FreeCAD.Vector(1,0,0),180), FreeCAD.Vector(0,0,0))
+		tmp = Draft.clone(corner)
+		group.addObject(tmp)
+		# First rotation.
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(0,0,0), FreeCAD.Rotation(FreeCAD.Vector(0,0,1),180), FreeCAD.Vector(0,0,0))
+		# Second rotation + shift.
+		tmp.Placement = FreeCAD.Placement(FreeCAD.Vector(self.LX,self.LY, self.LZ), FreeCAD.Rotation(FreeCAD.Vector(1,0,0),90), FreeCAD.Vector(0,0,0)).multiply(tmp.Placement)
+		
+	def create(self, convertToSolid):
+		self.checkDimensions()
+		group = self.document.addObject("App::DocumentObjectGroup", "frame box group")
+		self.createPipes(group, convertToSolid)
+		self.addCorners(group, convertToSolid)
+		return group
+
+
+class BoxFromTable:
+	"""Create a part with dimensions from a CSV table."""
+	def __init__ (self, document, pipe_table, corner_table):
+		self.document = document
+		self.pipe_table = pipe_table
+		self.corner_table = corner_table
+		self.LX = tu("12 in")
```

Comment:
Should there also be a 16 inch frame? That's the one I built because the [existing docs](http://opensourceecology.org/wiki/D3D_PVC_pipe_frame) mentioned using the same length of steel rod from the BOM (15.5 in)